tests/store.py

```diff
@@ -107,4 +113,55 @@ def assert_didnt_write(self, round: int):
             round (int): round to which the weights belong
 
         """
-        assert self.s3.writes[str(round)] == 0
+        assert self.s3.writes[f"{round}/global"] == 0
+
+
+class MockS3Participant(S3LocalWeightsWriter, S3GlobalWeightsReader):
+    """A partial mock of the ``xain_sdk.store.S3GlobalWeightsReader`` and
+    ``xain_sdk.store.S3LocalWeightsWriter`` class that does not
+    perform any IO. Instead, data is stored in memory.
+
+    """
+
+    def __init__(self, mock_s3_resource):
+        self.config = StorageConfig(
+            endpoint="endpoint",
+            access_key_id="access_key_id",
+            secret_access_key="secret_access_key",
+            bucket="bucket",
+        )
+        self.s3 = mock_s3_resource
+
+    def assert_wrote(self, participant_id: str, round: int, weights: np.ndarray):
+        """Check that the given weights have been written to the store for the
+        given round.
+
+        Args:
+            weights: weights to store
+            participant_id: ID of the participant
+            round: round to which the weights belong
+        """
+        key = f"{round}/{participant_id}"
+        writes = self.s3.writes[key]
+        assert writes == 1, f"got {writes} writes for {key}, expected 1"
+        assert_ndarray_eq(self.s3.fake_store[key], weights)
+
+    def assert_didnt_write(self, participant_id: str, round: int):
+        """Check that the weights for the given round have NOT been written to
+        the store.
+
+        Args:
+            participant_id: ID of the participant
+            round: round to which the weights belong
+
+        """
+        key = f"{round}/{participant_id}"
+        assert self.s3.writes[key] == 0
+
+
+def assert_ndarray_eq(nd_array1, nd_array2):
```

Comment:
this functionality (including handling of nan)  is already provided by `np.testing.assert_array_equal()`

Reply:
thanks, I removed this function.